From 4127d614ae7d4ff9dfa817b2d3cca1eff924beb9 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Sun, 4 Dec 2016 16:53:03 +0200
Subject: [PATCH 0805/1345] fix: dmatest: check for buffer boundary in source
 and destination pointers

commit  56e51b50f3cf666b165fd81d3f6d3b259fa75b20 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Current check did not take into acount a buffer override in case the total
buffer size is not a multiply of the test buffer size.

Change-Id: I83325bdf9438cec0aca96c3fe43496e756831686
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34347
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/dmatest.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/dmatest.c b/drivers/dma/dmatest.c
index 0ab5736..471b08f 100644
--- a/drivers/dma/dmatest.c
+++ b/drivers/dma/dmatest.c
@@ -539,7 +539,7 @@ static int dmatest_perf_func(void *data)
 			for (i = 0; i < src_cnt; i++) {
 				thread->srcs[i] = buf_pool_ptr;
 				buf_pool_ptr += params->buf_size;
-				if ((buf_pool_ptr - buf_pool_addr) >= buf_pool_size)
+				if ((buf_pool_ptr - buf_pool_addr + params->buf_size) >= buf_pool_size)
 					buf_pool_ptr = buf_pool_addr;
 			}
 		}
@@ -547,7 +547,7 @@ static int dmatest_perf_func(void *data)
 		for (i = 0; i < dst_cnt; i++) {
 			thread->dsts[i] = buf_pool_ptr;
 			buf_pool_ptr += params->buf_size;
-			if ((buf_pool_ptr - buf_pool_addr) >= buf_pool_size)
+			if ((buf_pool_ptr - buf_pool_addr + params->buf_size) >= buf_pool_size)
 				buf_pool_ptr = buf_pool_addr;
 		}
 
-- 
1.7.9.5

